### PR TITLE
Create F3X workflow step 2 page

### PR DESCRIPTION
**Pull request by mjtravers**
[app#133](https://github.com/fecgov/fecfile-web-app/issues/133)

### CODE REVIEW

File: front-end/package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
         "@popperjs/core": "^2.10.2",
         "bootstrap": "^5.1.3",
         "class-transformer": "^0.5.1",
-        "fecfile-validate": "https://github.com/fecgov/fecfile-validate#develop",
```
Comment (toddlees) [inline front-end/package.json:40]: I can't find this commit on the repo

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.html [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+              [value]="null"
+              label="NO"
+              formControlName="change_of_address"
+              ariaLabel="Address has not changed in the last 10 days"
+            ></p-radioButton>
+          </div>
+          <div>
+            <p-radioButton
+              name="change_of_address"
+              value="X"
```
Comment (toddlees) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.html:23]: Using the FECFormat data types is never going to sit well with me.  Not in scope, but maybe we can create some transformation layer in the validator code so we can use the correct datatypes in our code and contain the mess in the validator.  Making consolations like this and the date formatting i did in my pr, is going to keep adding up to a really unstable codebase

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.html [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+          <div>
+            <textarea [rows]="5" [cols]="30" pInputTextarea [autoResize]="true" formControlName="memo"></textarea>
+          </div>
+        </div>
+      </div>
+    </ng-container>
+    <p-divider></p-divider>
+    <div class="grid">
+      <div class="col-2">
+        <button pButton pRipple label="Back" icon="pi pi-times" class="p-button-info" (click)="save('back')"></button>
```
Comment (toddlees) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.html]: Do we want back to save?

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    private route: ActivatedRoute,
+    private reportService: F3xSummaryService,
+    private validateService: ValidateService,
+    private fb: FormBuilder,
+    private store: Store
+  ) {}
+
+  ngOnInit(): void {
+    // Refresh committee account details whenever page loads
+    this.store.dispatch(refreshCommitteeAccountDetailsAction());
```
Comment (toddlees) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts:52]: I hadn't thought of that scenario.  Nice catch

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        mergeMap((params): Observable<F3xSummary> => {
+          const id = Number(params.get('id'));
+          return this.reportService.get(id);
+        })
+      )
+      .subscribe((report: F3xSummary) => {
+        this.report = report;
+        this.form.patchValue({
+          change_of_address: 'not-selected',
+          street_1: this.report.street_1 ? this.report.street_1 : this.committeeAccount?.street_1,
```
Comment (toddlees) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts:74]: if you combine the committeeAccount and parammap observables, you dont need to create a `committeeAccount` component property.  combineLatest gives you the functionality of skipUntil and lines 37 and 55-57 all in one call

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    if (this.form.invalid) {
+      return;
+    }
+
+    const payload: F3xSummary = F3xSummary.fromJSON({
+      ...this.report,
+      ...this.validateService.getFormValues(this.form),
+    });
+
+    this.reportService.update(payload, this.formProperties).subscribe(() => {
```
Comment (toddlees) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts:106]: I didnt know this existed that's clean

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+      if (jump === 'continue' && this.report?.id) {
+        this.router.navigateByUrl(`/reports/f3x/create/step3/${this.report.id}`);
+      }
+      if (jump === 'back' && this.report?.id) {
```
Comment (toddlees) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts:110]: should saving take you to the list page?

**Review by toddlees — APPROVED** [review]

File: front-end/package.json (hunk 1 above)
Comment (mjtravers) [reply front-end/package.json:40]: Hmmmm. I will check in with you on this

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.html (hunk 2 above)
Comment (mjtravers) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.html:23]: Let's chat about this, too.

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.html (hunk 3 above)
Comment (mjtravers) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.html]: I ran it by Mitchell and he didn't say no. I assumed we want to save whenever we leave the page. A good question for Aurelia

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts (hunk 5 above)
Comment (mjtravers) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts:74]: Yeah, I like that. We need to wrap that code up into an ng resolver to port it across pages before load

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts (hunk 7 above)
Comment (mjtravers) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step2.component.ts:110]: Ah. I changed this. yep

**Pull request merged**